### PR TITLE
Filtering on end warranty should also allow to search future dates

### PR DESCRIPTION
**Pull request by liedekef**
Currently it is now possible to select future dates when wanting to see the warranty expiration dates of devices. This change helps to see which devices will need attention in the coming X time then.

### CODE REVIEW

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Thanks for the contribution.

Adding `'maybefuture' => true` to the search option declared in `Infocom::rawSearchOptionsToAdd()` would be a better solution, if it works.  Could you try it?

`Search` class is already way to complex and specific code should not be added anymore.

**Comment by liedekef** [comment]
I can confirm it works by adding 'maybefuture' => true to the search option declared in Infocom::rawSearchOptionsToAdd() . Do I cancel this PR and create a new one?

**Comment by cedric-anne** [comment]
> I can confirm it works by adding 'maybefuture' => true to the search option declared in Infocom::rawSearchOptionsToAdd() . Do I cancel this PR and create a new one?

You can just revert the first commit and push a new one on current PR. We will squash them when we will be ready to merge them.

**Comment by liedekef** [comment]
Change done

File: src/Search.php [hunk 1]
```diff
@@ -3384,9 +3384,8 @@ public static function displaySearchoptionValue($request = [])
                                     'text'  => __('Myself'),
                                 ]
                             ];
```
Comment (cedric-anne) [inline src/Search.php:3386]:
```suggestion
                            ];

```

File: src/Search.php [hunk 2]
```diff
@@ -3384,9 +3384,8 @@ public static function displaySearchoptionValue($request = [])
...
-
                             break;
-                    }
+                     }
```
Comment (cedric-anne) [inline src/Search.php:3388]:
```suggestion
                    }
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**